File: Data Structures & Algorithms/count-the-number-of-consistent-strings/submission-2.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        unordered_set<char> seen(allowed.begin(), allowed.end());
        int res  = 0;
        for (const string& word : words) {
            bool good = true;
            for (const auto& w : word) {
                if (seen.find(w) == seen.end()) {
                    good = false;
                    break;
                }
            }
            if (good == true) {
                res += 1;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/find-words-that-can-be-formed-by-characters/submission-2.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        hashmap = {}
        for ch in chars:
            hashmap[ch] = 1 + hashmap.get(ch, 0)
        res = 0
        for w in words:
            counter = Counter(w)
            good = True
            for c in counter:
                if counter[c] > hashmap.get(c, 0):
                    good= False
                    break
            if good:
                res += len(w)
        return res
                
            